wordpress-mobile/WordPress-iOS
# Update GoogleSignIn to 6.0.1

## Summary
This change updates `WordPressAuthenticator` dependency to `1.41.0-beta.2`, which upgrades `GoogleSignIn` to version `6.0.1`. For more details, please refer to #16924.

### References
- Closes #16924 
- https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/issues/609 
- https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/610

## To Test

- Ensure that the app can be compiled successfully. 
- Ensure that the Sign in with Google flow is not impacted:
  - Sign in with a Google account where **the email is associated with an existing WordPress account**. Verify that the flow is correct and not impacted by the change.
  - Sign in with a Google account where **the email is not associated with any WordPress account**. Verify that the flow is correct and not impacted by the change.
  - Sign in with a Google account **that has already been associated with a WordPress account**. Verify that the flow is correct and not impacted by the change.

## Regression Notes
1. Potential unintended areas of impact
Google sign in flow may be impacted.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested signing in with Google with two cases: (1) using a Google account with nonexistent email in WordPress, (2) using a Google account that corresponds to an existing WordPress account.

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
